clang/tools/clang-linker-wrapper/ClangLinkerWrapper.cpp

```diff
@@ -889,54 +890,41 @@ static Error runAOTCompileIntelGPU(SmallVector<std::string, 16> &InputFiles,
     CmdArgs.push_back(Arch);
   }
   addBackendOptions(Args, CmdArgs, /* IsCPU */ false);
-  for (unsigned I = 0; I < InputFiles.size(); ++I) {
-    const auto &File = InputFiles[I];
-    // Create a new file to write the translated file to.
-    auto TempFileOrErr =
-        createOutputFile(sys::path::filename(ExecutableName), "out");
-    if (!TempFileOrErr)
-      return TempFileOrErr.takeError();
-    CmdArgs.push_back("-output");
-    CmdArgs.push_back(*TempFileOrErr);
-    CmdArgs.push_back("-file");
-    CmdArgs.push_back(File);
-    if (Error Err = executeCommands(*OclocPath, CmdArgs))
-      return std::move(Err);
-    // Pop back last four items
-    CmdArgs.pop_back_n(4);
-    // Accumulate AOT output files (avoid for dry runs)
-    if (!DryRun)
-      OutputFiles[I] = *TempFileOrErr;
-  }
-  return Error::success();
+  // Create a new file to write the translated file to.
+  auto TempFileOrErr =
+      createOutputFile(sys::path::filename(ExecutableName), "out");
+  if (!TempFileOrErr)
+    return TempFileOrErr.takeError();
+  CmdArgs.push_back("-output");
+  CmdArgs.push_back(*TempFileOrErr);
+  CmdArgs.push_back("-file");
+  CmdArgs.push_back(InputFile);
+  if (Error Err = executeCommands(*OclocPath, CmdArgs))
+    return std::move(Err);
+  return *TempFileOrErr;
 }
 
-/// Run AOT compilation
-static Expected<StringRef> runAOTCompile(StringRef InputTable,
+/// Run AOT compilation for Intel CPU/GPU.
+/// 'InputFile' is the input SPIR-V file.
+/// 'Args' encompasses all arguments required for linking and wrapping device
+/// code and will be parsed to generate options required to be passed into the
+/// SYCL AOT compilation step.
+static Expected<StringRef> runAOTCompile(StringRef InputFile,
                                          const ArgList &Args) {
   const llvm::Triple Triple(Args.getLastArgValue(OPT_triple_EQ));
-  Table LiveSYCLTable;
-  if (Error Err = LiveSYCLTable.populateSYCLTable(InputTable))
-    return std::move(Err);
-  auto InputFiles = LiveSYCLTable.getListOfIRFiles();
-  // For dry runs, add a 'dummy' file to emit the command
-  if (DryRun)
-    InputFiles.push_back("dummy");
   if (Triple.isSPIRAOT()) {
-    SmallVector<std::string, 16> OutputFiles(InputFiles.size());
-    if (Triple.getSubArch() == llvm::Triple::SPIRSubArch_gen)
-      if (Error E = runAOTCompileIntelGPU(InputFiles, OutputFiles, Args))
-        return E;
-    if (Triple.getSubArch() == llvm::Triple::SPIRSubArch_x86_64)
-      if (Error E = runAOTCompileIntelCPU(InputFiles, OutputFiles, Args))
-        return E;
-    if (!DryRun)
-      for (int I = 0; I < OutputFiles.size(); ++I)
-        LiveSYCLTable.Entries[I].IRFile = OutputFiles[I];
-    auto Output = LiveSYCLTable.writeSYCLTableToFile();
-    if (!Output)
-      return Output.takeError();
-    return *Output;
+    if (Triple.getSubArch() == llvm::Triple::SPIRSubArch_gen) {
+      auto GPUOutputFile = runAOTCompileIntelGPU(InputFile, Args);
+      if (!GPUOutputFile)
```

Comment:
Do we really need to perform this step - can't we just `return runAOTCompileIntelGPU(InputFile, Args)` directly? We're just forwarding on a `Expected<StringRef>` from one function to another.

Reply:
Good point. I will update. Thanks